### ISSUE TITLE
Issue for isdisjoin()

### ISSUE THREAD

**Issue by FuzzyLogic** [qntm/greenery#36]
Hello,

I am seeing the following output.

>>> from greenery.lego import parse
>>> parse("/etc/.*").to_fsm().isdisjoint(parse("/etc/something.*").to_fsm())
True

However, I can verify that:
>>> parse("/etc/.*").to_fsm().accepts("/etc/something")
True
>>> parse("/etc/something.*").to_fsm().accepts("/etc/something")
True

Am I misinterpreting the semantics of the isdisjoint() method, or is this a bug.

Some background info: I have a similar motivation to yours, I'm interested in the intersection of two regular expressions. However, I am really only interested in knowing whether this intersection is empty. Is there a more efficient way to doing this than calculating 'parse(rx1) & parse(rx2)' and checking if it is empty? I am really trying to to check if two regular expressions that describe file system paths will overlap to some degree.

**Comment by qntm** [comment]
It looks as if Markdown has mangled your examples, but I think you were asking why this happens:

```
>>> parse("/etc/.*").to_fsm().isdisjoint(parse("/etc/something.*").to_fsm())
True
```

And yeah, that's a bug. Drilling down slightly we get:

```
>>> print(parse("/etc/.*").to_fsm() & parse("/etc/something.*").to_fsm())
  name final? / c e g h i m n o s t anything_else
--------------------------------------------------
* 0    False

>>> print(parse(".*").to_fsm() & parse("a.*").to_fsm())
  name final? a anything_else
------------------------------
* 0    False
```

Leave this with me...

**Comment by qntm** [comment]
Fixed with 984b060. Go get `greenery` version 3.1 to get the fixed version. Cheers!
